internal/services/recoveryservices/recovery_services_vault_resource.go

```diff
@@ -484,58 +429,6 @@ func resourceRecoveryServicesVaultUpdate(d *pluginsdk.ResourceData, meta interfa
 		},
 	}
 
-	if d.HasChanges("storage_mode_type", "cross_region_restore_enabled") {
```

Comment:
(same here)

Reply:
the storage update is done by the `updateAndPoll()`, it does not need a seperate client.